### PR TITLE
JDK-8266269:  Lookup::accessClass fails with IAE when accessing an arrayClass with a protected inner class as component class

### PR DESCRIPTION
**Pull request by mlchung**
`Lookup::accessClass` should determine the accessibility of the element type.  An array class is accessible if and only if its element type is accessible.

This also fixes a spec bug to document `@throws NullPointerException` if the argument is null.

Please review the CSR:
    https://bugs.openjdk.java.net/browse/JDK-8269312

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8266269](https://bugs.openjdk.java.net/browse/JDK-8266269): Lookup::accessClass fails with IAE when accessing an arrayClass with a protected inner class as component class

### Reviewers
 * [Chris Hegarty](https://openjdk.java.net/census#chegar) (@ChrisHegarty - **Reviewer**) ⚠️ Review applies to 96da75be490418f1d727583cae92429a3507c384
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk17 pull/137/head:pull/137` \
`$ git checkout pull/137`

Update a local copy of the PR: \
`$ git checkout pull/137` \
`$ git pull https://git.openjdk.java.net/jdk17 pull/137/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 137`

View PR using the GUI difftool: \
`$ git pr show -t 137`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk17/pull/137.diff">https://git.openjdk.java.net/jdk17/pull/137.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/lang/invoke/t8150782/TestFindClass.java [hunk 1]
```diff
@@ -24,19 +24,22 @@
...
 import java.lang.invoke.*;
+import p.Foo;
+import q.Bar;
 
 import static java.lang.invoke.MethodHandles.*;
 
 import static org.testng.AssertJUnit.*;
 
 import org.testng.annotations.*;
+import p.Foo;
```
Comment (ChrisHegarty) [inline test/jdk/java/lang/invoke/t8150782/TestFindClass.java:42]: maybe a stray import?

**Review by ChrisHegarty — APPROVED** [review]

File: test/jdk/java/lang/invoke/t8150782/TestFindClass.java (hunk 1 above)
Comment (mlchung) [reply test/jdk/java/lang/invoke/t8150782/TestFindClass.java:42]: Yup. Will remove it.

**Review by AlanBateman — APPROVED** [review]
Looks okay. In passing I think t8150782 is the only test directory using that naming convention. There are several tests using <bugid> in the directory name.

**Comment by mlchung** [comment]
I also notice that  `t8150782` outliner.   It'd actually prefer to remove the directory `t8150782` or  place the tests in a directory name with descriptive name.   That can be cleaned up as a separate issue.

**Pull request closed without merging**